Snap 517

rk-geode opened the pull request:
Done required changes to make hydra test runnable.

swatisawant committed:
- Added changes required for running the batterytests
- Hard-coded paths needs to be replaced at few places(will be doing that changes later on)
- Running batterytests also requires addition jars in classpath. I will add those jars at some shared location for time being.

Commit:
Removed hard coded path for genfirexd jar and jtest.
Made changes in hydra to use correct version of gemfirexd jar.
Added gradle task to copy bt,inc and conf files.

Commit:
Corrected build os path while copying conf,bt and inc files.

Commit:
Merge remote-tracking branch 'origin/snappy/master' into SNAP-517

Commit:
Add new run bt to run snappy-store regression.
Added sample runbt script (sample-runbt.sh) to run any bt by providing result directory and bt path as argument.
corrected the lib path for gemfirexd jar for hydra.

Commit:
Copying .sql scripts from test directory to build directory.
Addied jline and derbtnet jar in extraclasspath

Commit:
Merge remote-tracking branch 'origin/snappy/master' into SNAP-517

Commit:
Fixed for system.log path issue and classpath.

Commit:
Fixed jvm path.

swatisawant committed:
- Added support in the script to run the snappy hydra tests.
- Corrected the gemxd jar version. Currenlty it is picking up the wrong version.

Commit:
Merge branch 'snappy/master' into SNAP-517

Commit:
Merge branch 'snappy/master' into SNAP-517

Commit:
Merge branch 'snappy/master' into SNAP-517

Commit:
Merge remote-tracking branch 'origin/snappy/master' into SNAP-517

Commit:
Partial fix for SNAP-579.
Added gemfire product version for GemfireVersion.properties.
Using old derby version for hydra test.

Commit:
Did some cleanup.
Removed unwanted logs and hard coded path from conf files.

Commit:
Added jtest extra class path in eviction conf file.
Corrected syntex error issue in VersionDescription.java

Commit:
Modified derbyServerClassPath hydraprm to read expandble/extraclass path.

Commit:
Adding customer usecase for large volume test.

Commit:
Merge remote-tracking branch 'origin/snappy/master' into SNAP-517

Conflicts:
	tests/sql/build.gradle

Commit:
Removed unwanted files.
Correction made in mapper file to generate data

sumwale commented on build.gradle:
```diff
@@ -919,6 +919,40 @@ task product(dependsOn: [ subprojectBase + 'gemfirexd-client:jar',
+
+  copy {
+    def osDir = osName.getFamilyName().replace(" ", "").toLowerCase()
+    from (rootDir.getAbsolutePath() + '/snappy-store/tests/core/src/main/java') {
```
can use: project(subprojectBase + 'gemfire-tests').projectDir + '/src/main/java'

sumwale commented on build.gradle:
```diff
@@ -919,6 +919,40 @@ task product(dependsOn: [ subprojectBase + 'gemfirexd-client:jar',
+      include 'hydratest/hydracheck/*.sh'
+      include 'bin/scaleperf/*'
+    }
+    from (rootDir.getAbsolutePath() + '/snappy-store/tests/sql/src/main/java') {
```
can use: project(subprojectBase + 'gemfirexd-tests').projectDir + '/src/main/java'

sumwale commented on build.gradle:
```diff
@@ -919,6 +919,40 @@ task product(dependsOn: [ subprojectBase + 'gemfirexd-client:jar',
   }
+
+  copy {
+    def osDir = osName.getFamilyName().replace(" ", "").toLowerCase()
```
remove explicit osDir; use buildDir of the project as below

sumwale commented on build.gradle:
```diff
@@ -919,6 +919,40 @@ task product(dependsOn: [ subprojectBase + 'gemfirexd-client:jar',
+      include '**/*.ts'
+      include '**/*.pl'
+    }
+    into rootDir.getAbsolutePath() + '/snappy-store/tests/core/build-artifacts/' + osDir + '/classes/main'
```
use: project(subprojectBase + 'gemfirexd-tests').sourceSets.main.output.classesDir

Or: project(subprojectBase + 'gemfirexd-tests').buildDir + '/classes/main'

After removing gemfire-hydra-tests jar from extraClassPaths

sumwale commented on tests/sql/src/main/java/sql/snappy.local.conf:
```diff
@@ -0,0 +1,25 @@
+hydra.VmPrms-extraClassPaths        += $GEMFIRE/lib/gemfirexd-client-2.0-BETA.jar;
+hydra.VmPrms-extraClassPaths        += $GEMFIRE/lib/gemfirexd-tools-2.0-BETA.jar;
+hydra.VmPrms-extraClassPaths        += $GEMFIRE/lib/gemfirexd-2.0-BETA.jar;
+hydra.VmPrms-extraClassPaths        += $JTESTS/../../libs/gemfire-hydra-tests-2.0-BETA.jar;
```
Can remove gemfire-hydra-tests since it is already included in gemfirexd-hydra-tests below

sumwale commented:
See few comments above. Haven't looked at the new scripts, but will do that later and can go ahead with merge.

Commit:
Changes made as per Sumedh suggestions.
Updated jars version to 1.5-BETA to run the tests
Updated classpath to user gemxd-all jar.
